editor/writing-flow/index.js

```diff
@@ -45,6 +45,16 @@ class WritingFlow extends Component {
 		this.verticalRect = null;
 	}
 
+	componentDidMount() {
+		document.addEventListener( 'keydown', this.onKeyDown );
```

Comment:
Do we need these events to be monitored for the whole document? Or just `this.container` ?

Do we risk bugs with having `onKeyDown` now running for any keypress in the document?

Reply:
> Do we risk bugs with having onKeyDown now running for any keypress in the document?

Based on testing master, Yes.

Due to this change, there are massive issues with keyboard navigation. Essentially, pressing `up` or `down` anywhere will activate this code (even in the toolbar). That is because the react events are being bound to the `document` as well, so this is firing even if the component handlers themselves call `stopPropagation`.

There is a post explaining how this works: 
https://stackoverflow.com/questions/24415631/reactjs-syntheticevent-stoppropagation-only-works-with-react-events

This is forcing component developers to write `event.nativeEvent.stopPropagation`, otherwise their event handler AND this handler will fire. And WritingFlow is pretty much always on the screen, so this handler interferes with almost everything. 

Now that we have UI testing, we need to have some basic user navigation tests setup so this kind of thing will be easier to catch in the future.

Reply:
😱